<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Meshing_dialog</class>
 <widget class="QDialog" name="Meshing_dialog">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>568</width>
    <height>1049</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Meshing criteria</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item>
    <widget class="QLabel" name="objectName">
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>NO OBJECT</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="objectNameSize">
     <property name="text">
      <string>No size</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="initializationGroup">
     <property name="title">
      <string>Mesh initialization</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5" columnstretch="0,1,0">
      <item row="0" column="0">
       <widget class="QLabel" name="detectComponentsLabel">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>&amp;Detect all connected components</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>detectComponents</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QCheckBox" name="detectComponents">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="sharpFeaturesGroup">
     <property name="layoutDirection">
      <enum>Qt::LeftToRight</enum>
     </property>
     <property name="title">
      <string>Sharp features</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0">
      <item row="3" column="1">
       <widget class="DoubleEdit" name="sharpEdgesAngle">
        <property name="text">
         <string>60.00</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="protect">
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="sharpEdgesAngleLabel">
        <property name="toolTip">
         <string>For the detection of sharp edges, this is the lower bound on the angle between normals of incident facets of sharp edges</string>
        </property>
        <property name="text">
         <string>Angle bound for &amp;sharp edges (deg)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>sharpEdgesAngle</cstring>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <widget class="QCheckBox" name="noEdgeSizing">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="protectEdges"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="protectLabel">
        <property name="text">
         <string>&amp;Protect sharp edges</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>protect</cstring>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="DoubleEdit" name="edgeSizing">
        <property name="text">
         <string>0.0</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="edgeLabel">
        <property name="text">
         <string>Edge &amp;size</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>edgeSizing</cstring>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Surface</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0,0,0,0,0,0,0,0" columnstretch="0,1,0">
      <item row="2" column="2">
       <widget class="QCheckBox" name="noFacetSizing">
        <property name="toolTip">
         <string>Enable/Disable parameter</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="approxLabel">
        <property name="text">
         <string>Approximation &amp;error</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>approx</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="noApprox">
        <property name="toolTip">
         <string>Enable/Disable parameter</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="angleLabel">
        <property name="text">
         <string>Facet min. &amp;angle (deg)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>facetAngle</cstring>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="3">
       <widget class="QFrame" name="advanced">
        <property name="visible">
         <bool>true</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <layout class="QVBoxLayout" name="advancedLayout">
         <item>
          <widget class="QLabel" name="manifoldLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>The following option forces the meshing engine to produce a 3D mesh with a manifold boundary. That may lead to an infinite loop if the input domain has a non-manifold 2D component.</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="manifoldCheckBox">
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
           <property name="text">
            <string>&amp;manifold</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="facetTopologyLabel">
           <property name="text">
            <string>The following option enables the &lt;a href=&quot;https://doc.cgal.org/latest/Mesh_3/group__PkgMesh__3Enum.html#gaac682a581ddebf8daff681621911d1f3&quot;&gt; &lt;code&gt;FACET_VERTICES_ON_SAME_SURFACE_PATCH_WITH_ADJACENCY_CHECK &lt;/code&gt;&lt;/a&gt; facet topology criterion (click on the link to see the manual page).</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="facetTopology">
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
           <property name="text">
            <string>force facets vertices on same surface patch</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="checkBox">
        <property name="text">
         <string>See advanced &amp;controls...</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QCheckBox" name="noAngle">
        <property name="toolTip">
         <string>Enable/Disable parameter</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="sizingLabel">
        <property name="text">
         <string>&amp;Facet max. size</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>facetSizing</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="DoubleEdit" name="facetSizing">
        <property name="text">
         <string>0.065</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="DoubleEdit" name="facetAngle">
        <property name="inputMask">
         <string/>
        </property>
        <property name="text">
         <string>25.00</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="DoubleEdit" name="approx">
        <property name="text">
         <string>0.0017</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="volumeGroup">
     <property name="title">
      <string>Volume</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,1,0">
      <item row="0" column="2">
       <widget class="QCheckBox" name="noTetSizing">
        <property name="toolTip">
         <string>Enable/Disable parameter</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="noTetShape">
        <property name="toolTip">
         <string>Enable/Disable parameter</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Tetrahedron shape (radius-edge ratio)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>&amp;Tetrahedron max. size</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="buddy">
         <cstring>tetSizing</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="DoubleEdit" name="tetSizing">
        <property name="text">
         <string>0.00</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="DoubleEdit" name="tetShape">
        <property name="text">
         <string>3.00</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grayImgGroup">
     <property name="title">
      <string>Gray Images</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4" columnstretch="0,1">
      <item row="2" column="1">
       <widget class="QCheckBox" name="inside_is_less_checkBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>if unchecked, value inside the domain is greater than isovalue</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="toolTip">
         <string>image value outside the image bounding box</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Value outside</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Iso value</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Value inside is less than iso value</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="DoubleEdit" name="value_outside_spinBox">
        <property name="text">
         <string>0.00</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="DoubleEdit" name="iso_value_spinBox">
        <property name="text">
         <string>3.00</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="labeledImgGroup">
     <property name="title">
      <string>Labeled Images</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="0" column="0">
       <widget class="QLabel" name="useWeights_label">
        <property name="layoutDirection">
         <enum>Qt::RightToLeft</enum>
        </property>
        <property name="text">
         <string>Use Weighted Image</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="useWeights_checkbox">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="tristate">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="weightsSigma_label">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>standard deviation parameter of the internal Gaussian filter</string>
        </property>
        <property name="text">
         <string>Sigma </string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="DoubleEdit" name="weightsSigma">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="facegraphCheckBox">
     <property name="text">
      <string>Create as FaceGraph</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DoubleEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">CGAL_double_edit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>detectComponents</tabstop>
  <tabstop>protectEdges</tabstop>
  <tabstop>protect</tabstop>
  <tabstop>sharpEdgesAngle</tabstop>
  <tabstop>edgeSizing</tabstop>
  <tabstop>noEdgeSizing</tabstop>
  <tabstop>approx</tabstop>
  <tabstop>noApprox</tabstop>
  <tabstop>facetSizing</tabstop>
  <tabstop>noFacetSizing</tabstop>
  <tabstop>facetAngle</tabstop>
  <tabstop>noAngle</tabstop>
  <tabstop>tetSizing</tabstop>
  <tabstop>noTetSizing</tabstop>
  <tabstop>tetShape</tabstop>
  <tabstop>noTetShape</tabstop>
  <tabstop>value_outside_spinBox</tabstop>
  <tabstop>iso_value_spinBox</tabstop>
  <tabstop>inside_is_less_checkBox</tabstop>
  <tabstop>useWeights_checkbox</tabstop>
  <tabstop>weightsSigma</tabstop>
  <tabstop>facegraphCheckBox</tabstop>
  <tabstop>checkBox</tabstop>
  <tabstop>manifoldCheckBox</tabstop>
  <tabstop>facetTopology</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Meshing_dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>403</x>
     <y>655</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Meshing_dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>403</x>
     <y>655</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>advanced</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>339</y>
    </hint>
    <hint type="destinationlabel">
     <x>78</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
